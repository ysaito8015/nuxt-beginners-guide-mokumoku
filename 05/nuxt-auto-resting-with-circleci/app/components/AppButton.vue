<template>
  <div>
    <p
      v-if="clicked"
      >
      clicked
    </p>
    <button
      type="button"
      @click="handleClick"
      >
      click
    </button>
  </div>
</template>

<script>
export default {
  data() {
    return {
      clicked: false
    }
  },
  methods: {
    handleClick() {
      this.clicked = true
    }
  }
}
</script>
