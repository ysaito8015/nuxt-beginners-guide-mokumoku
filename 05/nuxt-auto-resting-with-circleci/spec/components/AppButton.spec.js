import { mount } from '@vue/test-utils'
import AppButton from '~/components/AppButton.vue'

describe('components/AppBUtton.vue', () => {
  test('match snapshot(unclicked)', () => {
    const wrapper = mount(AppButton)
    expect(wrapper.element).toMatchSnapshot()
  })
  test('match snapshot(clicked)', () => {
    const wrapper = mount(AppButton)
    wrapper.findComponent('button').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })
})
