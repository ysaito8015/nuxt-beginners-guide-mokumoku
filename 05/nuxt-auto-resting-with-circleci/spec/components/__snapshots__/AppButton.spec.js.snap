// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AppBUtton.vue match snapshot(clicked) 1`] = `
<div>
  <!---->
   
  <button
    type="button"
  >
    
    click
  
  </button>
</div>
`;

exports[`components/AppBUtton.vue match snapshot(unclicked) 1`] = `
<div>
  <!---->
   
  <button
    type="button"
  >
    
    click
  
  </button>
</div>
`;
